configure_cmd.go

```diff
@@ -0,0 +1,130 @@
+// provided URL is URL of Satellite server.
+//
+// We would like to use different approach in the future. We would like to use
+// some API endpoints not restricted by username & password for getting CA certs
+// and rendered rhsm.conf, because it would be more secure, but it is not possible ATM
+func satelliteAction(ctx *cli.Context) error {
+	var configureSatelliteResult ConfigureSatelliteResult
+	configureSatelliteResult.format = ctx.String("format")
+	satelliteUrlStr := ctx.String("url")
+
+	uid := os.Getuid()
+	if uid != 0 {
+		errMsg := "non-root user cannot connect system"
+		exitCode := 1
+		return cli.Exit(fmt.Errorf("error: %satSpinner", errMsg), exitCode)
+	}
+
+	hostname, err := os.Hostname()
+	if uiSettings.isMachineReadable {
+		configureSatelliteResult.Hostname = hostname
+	}
+	if err != nil {
+		exitCode := 1
+		if uiSettings.isMachineReadable {
+			configureSatelliteResult.HostnameError = err.Error()
+			return cli.Exit(configureSatelliteResult, exitCode)
+		} else {
+			return cli.Exit(err, exitCode)
+		}
+	}
+
+	satelliteUrl, err := normalizeSatelliteScriptUrl(satelliteUrlStr)
+	if err != nil {
+		return cli.Exit(fmt.Errorf("could not parse satellite url: %w", err), 1)
+	}
+
+	configureSatelliteResult.SatelliteServerHostname = satelliteUrl.Hostname()
+	configureSatelliteResult.SatelliteServerScriptUrl = satelliteUrl.String()
+
+	var satSpinner *spinner.Spinner = nil
+	if uiSettings.isRich {
+		satSpinner = spinner.New(spinner.CharSets[9], 100*time.Millisecond)
+		satSpinner.Suffix = fmt.Sprintf(" Configuring '%v' to use Satellite %v", hostname, satelliteUrl.Host)
+		satSpinner.Start()
+		// Stop spinner after running function
+		defer func() { satSpinner.Stop() }()
+	}
+
+	err = pingSatelliteServer(satelliteUrl, satSpinner)
+	if err != nil {
+		return cli.Exit(fmt.Errorf("unable to verify that given server is Satellite server: %v", err), 1)
+	}
+
+	configureSatelliteResult.IsServerSatellite = true
+
+	// Create file for script file
+	satelliteScriptPath := filepath.Join(VarLibDir, "katello-rhsm-consumer")
+	defer os.Remove(satelliteScriptPath)
+	scriptFile, err := os.Create(satelliteScriptPath)
+	if err != nil {
+		return cli.Exit(fmt.Errorf("could not create %v file: %w", satelliteScriptPath, err), 1)
+	}
+	defer scriptFile.Close()
+	err = os.Chmod(satelliteScriptPath, 0700)
+	if err != nil {
+		return cli.Exit(fmt.Errorf("could not set permissions on %v file: %w", satelliteScriptPath, err), 1)
+	}
+
+	err = downloadSatelliteScript(scriptFile, satelliteUrl, satSpinner)
+	if err != nil {
+		return cli.Exit(fmt.Errorf("could not download satellite bootstrap script: %w", err), 1)
+	}
+
+	if satSpinner != nil {
+		satSpinner.Suffix = fmt.Sprintf(" Configuring '%v' to use Satellite server: %v", hostname, satelliteUrl.Host)
+	}
+	// Run the script. It should install CA certificate, change configuration of rhsm.conf.
+	// In theory, it can do almost anything.
+	cmd := exec.Command(satelliteScriptPath)
+	err = cmd.Run()
```

Comment:
I'm thinking ahead here toward the ability to troubleshoot this execution when
things go wrong. Should we log the stdout and stderr of the run to a file
somewhere? I don't think it's necessary to pipe the stdout and stderr into this
parent process, but logging the script's output might be helpful.

Reply:
I agree that it would be useful to log more information from `rhc`. Problem is that we do not have any logging system for `rhc`. Should we address this issue after solving this issue https://github.com/RedHatInsights/rhc/issues/88 ?

Reply:
I agree that deferring it out of this PR is the right decision, since this PR is
already sizable. But I'm concerned that we won't prioritize it enough in a
followup PR.